### PR TITLE
Optimize stories

### PR DESCRIPTION
**Pull request by lightalloy**
## What type of PR is this? (check all applicable)
- [x] Refactor

## Description
- Added index to articles path, since queries by path are popular when retrieving articles
- select only needed columns for tag colors
- Make fewer SQL queries on retrieving sticky article collection

### CODE REVIEW

File: app/labor/sticky_article_collection.rb [hunk 1]
```diff
@@ -18,16 +17,15 @@ def user_stickies
...
   end
 
   def tag_articles
-    Article.tagged_with(article_tags, any: true).
+    @tag_articles ||= Article.tagged_with(article_tags, any: true).
       includes(:user).
       where("positive_reactions_count > ? OR comments_count > ?", reaction_count_num, comment_count_num).
       where(published: true).
       where.not(id: article.id, user_id: article.user_id).
-      limited_column_select.
```
Comment (lightalloy) [inline app/labor/sticky_article_collection.rb]: Removed `limited_column_select` cause is wasn't helpful, cause `tagged_with` still adds `articles.*` to `select` columns list.

**Comment by lightalloy** [comment]
Can't get what's the problem at codeclimate and Travis with the js files I haven't touched. Locally tests run mostly fine (I get several other errors both on master and my branch when I run `yarn run jest`)

**Comment by rhymes** [comment]
It looks like a bug in the jest-util library...

**Comment by Zhao-Andy** [comment]
Hm, rerunning the build. For posterity, all tests had this error message:
image 2019-03-07 at 8 31 22 am

**Comment by lightalloy** [comment]
https://github.com/facebook/jest/issues/8069
Probably related to this issue.

**Comment by Zhao-Andy** [comment]
@maestromac probably safe to upgrade Jest to `latest` to resolve this right? We are on 23.5.0 and latest is 24.3.0.

**Comment by maestromac** [comment]
Jest 24 won't work for us https://github.com/thepracticaldev/dev.to/pull/1744#issuecomment-464168349

**Comment by lightalloy** [comment]
@maestromac thanks for providing a solution for that.

**Pull request merged**